crypto/keys/src/ed25519.rs

```diff
@@ -450,16 +479,20 @@ impl<'a> TryFrom<&'a [u8]> for Ed25519Signature {
     }
 }
 
-impl TryFrom<Vec<u8>> for Ed25519Signature {
+#[cfg(feature = "alloc")]
+impl TryFrom<alloc::vec::Vec<u8>> for Ed25519Signature {
     type Error = SignatureError;
 
-    fn try_from(src: Vec<u8>) -> Result<Self, Self::Error> {
+    fn try_from(src: alloc::vec::Vec<u8>) -> Result<Self, Self::Error> {
         src.as_slice().try_into()
     }
 }
 
 derive_repr_bytes_from_as_ref_and_try_from!(Ed25519Signature, U64);
+
+#[cfg(feature = "prost")]
 derive_prost_message_from_repr_bytes!(Ed25519Signature);
+
 derive_core_cmp_from_as_ref!(Ed25519Signature);
 derive_debug_and_display_hex_from_as_ref!(Ed25519Signature);
```

Comment:
Optional/Equivalent: Move the conditional derive last
```suggestion
derive_core_cmp_from_as_ref!(Ed25519Signature);
derive_debug_and_display_hex_from_as_ref!(Ed25519Signature);

#[cfg(feature = "prost")]
derive_prost_message_from_repr_bytes!(Ed25519Signature);
```